openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -2333,14 +2685,22 @@ def check_handler_compatibility(self, other_handler):
         IncompatibleParameterError if handler_kwargs are incompatible with existing parameters.
         """
         string_attrs_to_compare = [
-            "potential",
             "fractional_bondorder_method",
             "fractional_bondorder_interpolation",
         ]
         self._check_attributes_are_equal(
             other_handler, identical_attrs=string_attrs_to_compare
         )
 
+        # potential="harmonic" and potential="(k/2)*(r-length)^2" should be considered identical
+        self_has_harmonic_potential = self.potential == 'harmonic' or self.potential == "(k/2)*(r-length)^2"
+        other_has_harmonic_potential = other_handler.potential == 'harmonic' or other_handler.potential == "(k/2)*(r-length)^2"
+        if not(self_has_harmonic_potential and other_has_harmonic_potential):
+            if self.potential != other_handler.potential:
+                raise IncompatibleParameterError(
+                    f"potential values are not identical. "
+                    f"(handler value: {self.potential}, incompatible value: {other_handler.potential}")
```

Comment:
Worth considering in other spec discussions if `harmonic` should be deprecated at some point after a larger push to explicitly encode expressions

Reply:
I'd advocate for silent upconversion over deprecation, but thematically yes, I'd like the word "harmonic" to stop appearing in our FFs.